### ISSUE TITLE
Using  { subscribe: true }  with useFirestoreQueryData causes infinite loading status.

### ISSUE THREAD

**Issue by atanaskanchev** [invertase/react-query-firebase#23]
Using  { subscribe: true }  with useFirestoreQueryData or useFirestoreQuery causes infinite loading status when I refresh a page in Nextjs app.

```javascript
const ContactsLandingPage: React.FC = (): React.ReactElement => {
  // ? next router
  const { push } = useRouter()

  // ? permissions gate
  const {
    customClaims: { accountId }
  } = usePermissionsGateContext()

  // ? the firebase services
  const firestore = useFirestore()

  // ? the contacts
  const contactsColRef = collection(firestore, `customer_accounts_data/${accountId}/contact`).withConverter(
    contactConverter
  )
  const {
    status: contactsStatus,
    error: contactsError,
    data: contacts
  } = useFirestoreQueryData(['contacts', { accountId }], contactsColRef, { subscribe: true }, {})

```
    "firebase": "^9.4.0",
    "@react-query-firebase/firestore": "^0.4.2",
    "react-query": "^3.32.3",

![image](https://user-images.githubusercontent.com/4958587/141699668-abf430e4-76a2-48b9-8d55-c4bcbfd0d9f8.png)

**Comment by atanaskanchev** [comment]
New findings

The "permissions gate" above was using `useAuthIdToken` to get the custom claims

```javascript
export default function useIdTokenResult() {
  const auth = getAuth(useFirebaseApp())
  return useAuthIdToken(['token'], auth, { forceRefresh: true })
}
```
Changing forceRefresh to false solved the issue, however, this doesn't explain why it got stuck in fetching in the first place.

**Comment by Ehesp** [comment]
Oh weird, not seem this myself (or in tests). Is there anything showing in console when the request is hanging?

**Comment by atanaskanchev** [comment]
Hi @Ehesp. There's no error in the console. The edge case here is that this happens only on page refresh. The initial load works just fine, but if I refresh the page the query remains stuck in fetching state indefinitely.

**Comment by Ehesp** [comment]
What is the difference between a page refresh and initial load? Also I assume this works ok when subscribe is false?

**Comment by atanaskanchev** [comment]
Hi @Ehesp  

The first load if `{ forceRefresh: true }`  works as expected,

![image](https://user-images.githubusercontent.com/4958587/142040324-bb01c384-08eb-4c5e-800d-595cbba26476.png)

however after F5 all the queries having `{ subscribe: true }` hand in indefinite fetching state 

![image](https://user-images.githubusercontent.com/4958587/142040373-dc195f10-593c-4a2a-ab08-d73a74fba739.png)

**Comment by Ehesp** [comment]
Huh, I'm confused :D Are you doing any sort of local caching? I'd of thought a page refresh would reset everything anyway back to square one. 

I'll see if I can reproduce this.

**Comment by atanaskanchev** [comment]
@Ehesp  The issue was resolved in the latest ```"react-query": "^3.33.7"```, couldn't pinpoint which commit fixed it though. Closing
